ironfish/src/rpc/server.ts

```diff
@@ -2,24 +2,27 @@
  * License, v. 2.0. If a copy of the MPL was not distributed with this
  * file, You can obtain one at https://mozilla.org/MPL/2.0/. */
 
+import { InternalStore } from '../fileStores'
 import { createRootLogger, Logger } from '../logger'
-import { IronfishNode } from '../node'
 import { IRpcAdapter } from './adapters'
-import { ApiNamespace, Router, router } from './routes'
+import { ApiNamespace, RequestContext, Router, router } from './routes'
 
 export class RpcServer {
-  readonly node: IronfishNode
+  readonly internal: InternalStore
+  readonly context: RequestContext
   readonly adapters: IRpcAdapter[] = []
 
-  private readonly router: Router
   private _isRunning = false
   private _startPromise: Promise<unknown> | null = null
   logger: Logger
 
-  constructor(node: IronfishNode, logger: Logger = createRootLogger()) {
-    this.node = node
-    this.router = router
-    this.router.server = this
+  constructor(
```

Comment:
I think the biggest change architectually here is you're now tying this to the root router in the router file. I guess that's ok, I just prefer not to have any module directly ever depend on a module level global since they are different concepts. I would prefer to still pass this in. What if I want to spawn a new RPC server from a non root level router that's already been filtered?

Reply:
Not sure I understand. The previous code was setting `server.router` to the root router. This code is just doing the same thing. I agree we could separate it more and not depend on the root router but I don't see how this is changing anything from current code.

Reply:
I might be misunderstanding, but it looks like they are already explicitly tied together -- the server sets its router property to the root router in the constructor